lib/common-request.js

```diff
@@ -28,14 +28,31 @@ const privates = require('./private-map');
 const request = require('request');
 
 module.exports = function doRequest (client, options) {
+  const clientConfig = privates.get(client).config;
+  if (clientConfig.user.token !== undefined) {
+    return doRequestWithToken(client, options, clientConfig.user.token);
+  } else {
+    return doRequestWithBasicAuth(client).then((accessToken) => {
```

Comment:
you also need to pass the options here.  i was getting a self-sign cert error from the request library even though i was passing in the "request options"

look at the doRequestWithToken function

Reply:
Oh yes I see. Thanks Luke. I'll update it.

Reply:
@lholmquist Hmm I am not sure TBH. Did you mean somthing like https://github.com/tremes/openshift-rest-client/commit/22b65f18e89cc45ff22f4e9b9f0390c73bc749bf ?

Reply:
@tremes very close,  you also need the requests settings to be merged,  something like this https://github.com/lholmquist/openshift-rest-client/commit/20bfbab219819892f291de29f86dd51233239938

i combined the headers and the base options in that commit,  but i like what you did separating them instead :)

Reply:
@lholmquist ahh ok thanks. So I updated it like this https://github.com/tremes/openshift-rest-client/commit/ca4f990efc9678b3dc0fc8e5db0e51046c9bb7b2. Will cherry-pick and rebase this PR if it's ok.

Reply:
no problem

Reply:
Done. Thanks. I am looking forward to a new release:-)